src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/ValidationError.cs

```diff
@@ -236,5 +243,41 @@ public ValidationError AddStackFrame(StackFrame stackFrame)
             StackFrames.Add(stackFrame);
             return this;
         }
+
+        /// <summary>
+        /// Adds the current stack frame to the list of stack frames and returns the updated object.
+        /// If there is no cache entry for the given file path and line number, a new stack frame is created and added to the cache.
+        /// </summary>
+        /// <param name="filePath">The path to the file from which this method is called. Captured automatically by default.</param>
+        /// <param name="lineNumber">The line number from which this method is called. CAptured automatically by default.</param>
+        /// <param name="skipFrames">The number of stack frames to skip when capturing. Used to avoid capturing this method and get the caller instead.</param>
+        /// <returns>The updated object.</returns>
+        internal ValidationError AddCurrentStackFrame([CallerFilePath] string filePath = "", [CallerLineNumber] int lineNumber = 0, int skipFrames = 1)
+        {
+            // We add 1 to the skipped frames to skip the current method
```

Comment:
If `skipFrames` defaults to `1` and you add `1` to it, is this comment accurate?

Reply:
Say we call:
DoSomething(), which calls
AddCurrentStackFrame(), which calls
GetCurrentStackFrame(), which instantiates
new StackFrame(true).

`GetCurrentStackFrame()` passes the 1 skip frame to skip itself from the stack and get the method that called it, which is the normal behaviour for any cases where we get the stack frame and use it, say to instantiate a new `ValidationError`.
But when using `AddCurrentStackFrame(()`, we need to skip one more layer to account for both `GetCurrentStackFrame()` and `AddCurrentStackFrame()`, and only capture the caller to this method.

Currently, we use the two methods for different situations:
- `GetCurrentStackFrame()` is used when we are creating a new `ValidationError` and we are pointing at where that occurs.
- `AddCurrentStackFrame()` is used when we have an existing `ValidationError` with some stack frames in it, to add the next stack frame and create the right chain for the stack trace.

You can see an example of this in the test class used in the test.